api/handlers/record_handler.go

```diff
@@ -0,0 +1,239 @@
+}
+
+func (handler *RecordHandler) GetByType(w http.ResponseWriter, r *http.Request) {
+	name := mux.Vars(r)["name"]
+	typ, err := validateType(name)
+	if err != nil {
+		handler.log.WithField("type", name).
+			Error(err)
+		writeJSONError(w, http.StatusNotFound, err.Error())
+		return
+	}
+
+	recordRepo, err := handler.recordRepositoryFactory.For(typ)
+	if err != nil {
+		handler.log.WithField("type", typ).
+			Errorf("error constructing record repository: %v", err)
+		status, message := handler.responseStatusForError(err)
+		writeJSONError(w, status, message)
+		return
+	}
+	filterCfg := filterConfigFromValues(r.URL.Query())
+
+	records, err := recordRepo.GetAll(r.Context(), filterCfg)
+	if err != nil {
+		handler.log.WithField("type", typ).
+			Errorf("error fetching records: GetAll: %v", err)
+		status, message := handler.responseStatusForError(err)
+		writeJSONError(w, status, message)
+		return
+	}
+
+	fieldsToSelect := handler.parseSelectQuery(r.URL.Query().Get("select"))
+	if len(fieldsToSelect) > 0 {
+		records = handler.selectRecordFields(fieldsToSelect, records)
+	}
+	writeJSON(w, http.StatusOK, records)
+}
+
+func (handler *RecordHandler) GetOneByType(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	var (
+		typeName = vars["name"]
+		recordID = vars["id"]
+	)
+
+	typ, err := validateType(typeName)
+	if err != nil {
+		handler.log.WithField("type", typeName).
+			Error(err)
+		writeJSONError(w, http.StatusNotFound, err.Error())
+		return
+	}
+
+	recordRepo, err := handler.recordRepositoryFactory.For(typ)
+	if err != nil {
+		handler.log.WithField("type", typeName).
+			Errorf("internal: error construing record repository: %v", err)
+
+		status := http.StatusInternalServerError
+		writeJSONError(w, status, http.StatusText(status))
+		return
+	}
+
+	record, err := recordRepo.GetByID(r.Context(), recordID)
+	if err != nil {
+		handler.log.WithField("type", typeName).
+			WithField("record", recordID).
+			Errorf("error fetching record: %v", err)
+
+		status, message := handler.responseStatusForError(err)
+		writeJSONError(w, status, message)
+		return
+	}
+	writeJSON(w, http.StatusOK, record)
+}
+
+func (handler *RecordHandler) parseSelectQuery(raw string) (fields []string) {
+	tokens := strings.Split(raw, ",")
+	for _, token := range tokens {
+		token = strings.TrimSpace(token)
+		if token == "" {
+			continue
+		}
+		fields = append(fields, token)
+	}
+	return
+}
+
+func (handler *RecordHandler) selectRecordFields(fields []string, records []record.Record) (processedRecords []record.Record) {
+	for _, record := range records {
+		newData := map[string]interface{}{}
+		for _, field := range fields {
+			v, ok := record.Data[field]
+			if !ok {
+				continue
+			}
+			newData[field] = v
+		}
+		record.Data = newData
+		processedRecords = append(processedRecords, record)
+	}
+	return
+}
+
+func (handler *RecordHandler) validateRecord(record record.Record) error {
```

Comment:
+1 on we should use a validation library to handle this, but it may not be on this PR for these following reasons:
- This is a legacy code which I don't want to mess with while package restructuring. (despite it is a new file, actually it is just a file rename from `api/handlers/type_handler.go`
- We will change Columbus to use [grcp-gateway](https://github.com/grpc-ecosystem/grpc-gateway) as its http and grpc handler. Once we move to that it already has its suggested validation libraries that we are already using across our odpf services e.g. [github.com/envoyproxy/protoc-gen-validate](https://github.com/envoyproxy/protoc-gen-validate).